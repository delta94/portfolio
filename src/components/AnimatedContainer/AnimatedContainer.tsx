import React from 'react';
import { animated, useSpring } from 'react-spring';

interface ThreeDAnimationProps {}

const calc: (x: number, y: number) => number[] = (x: number, y: number) => [
  -(y - window.innerHeight / 2) / 20,
  (x - window.innerWidth / 2) / 20,
  1.1,
];
const trans: (x: number, y: number, s: number) => string | undefined = (x: number, y: number, s: number) =>
  `perspective(600px) rotateX(${x}deg) rotateY(${y}deg) scale(${s})`;

const ThreeDAnimation: React.FC<ThreeDAnimationProps> = ({ children }) => {
  const [props, set] = useSpring(() => ({ xys: [0, 0, 1], config: { mass: 5, tension: 350, friction: 40 } }));

  return (
    <animated.div
      onMouseMove={({ clientX: x, clientY: y }) => set({ xys: calc(x, y) })}
      onMouseLeave={() => set({ xys: [0, 0, 1] })}
      style={{ transform: props.xys.interpolate(trans as any) }}
    >
      {children}
    </animated.div>
  );
};

export { ThreeDAnimation };
