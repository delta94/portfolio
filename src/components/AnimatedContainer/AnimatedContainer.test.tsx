import { render } from '@testing-library/react';
import React from 'react';
import { TestContextProvider } from '../../TestContextProvider';
import { ThreeDAnimation, VanishAnimation } from './AnimatedContainer';

describe('AnimatedContainer component render correctly', () => {
  test('ThreeDAnimation renders correctly', () => {
    const { asFragment } = render(
      <TestContextProvider>
        <ThreeDAnimation> mock children</ThreeDAnimation>
      </TestContextProvider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('VanishAnimation renders correctly', () => {
    const { asFragment } = render(
      <TestContextProvider>
        <VanishAnimation words={['mock', 'words']} />
      </TestContextProvider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
