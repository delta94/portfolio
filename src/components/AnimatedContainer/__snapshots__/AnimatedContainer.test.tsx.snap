// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimatedContainer component render correctly ThreeDAnimation renders correctly 1`] = `
<DocumentFragment>
  <div
    style="transform: perspective(600px) rotateX(0deg) rotateY(0deg) scale(1);"
  >
     mock children
  </div>
</DocumentFragment>
`;

exports[`AnimatedContainer component render correctly VanishAnimation renders correctly 1`] = `
<DocumentFragment>
  <div
    style="opacity: 0; transform: translate3d(0,20px,0);"
  >
    <h2
      class="h2Text-0-2-1"
    >
      mock
    </h2>
  </div>
  <div
    style="opacity: 0; transform: translate3d(0,20px,0);"
  >
    <h2
      class="h2Text-0-2-1"
    >
      words
    </h2>
  </div>
</DocumentFragment>
`;
